### PR TITLE
switch order of time series output

### PR DESCRIPTION
**Pull request by agibsonccc**
## What changes were proposed in this pull request?
Change dimensions of time series output.
(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

Please review
https://github.com/deeplearning4j/deeplearning4j/blob/master/CONTRIBUTING.md before opening a pull request.

### CODE REVIEW

**Comment by agibsonccc** [comment]
@emsixteeen @AlexDBlack

File: datavec-api/src/main/java/org/datavec/api/timeseries/util/TimeSeriesWritableUtils.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+            }
+        }
+
+
+        arr = Nd4j.create(new int[] {details.getMinValues(), size, details.getMaxTSLength()}, 'f');
+
+        boolean needMaskArray = false;
+        for (List<List<Writable>> c : list) {
+            if (c.size() < details.getMaxTSLength())
+                needMaskArray = true;
```
Comment (AlexDBlack) [inline datavec-api/src/main/java/org/datavec/api/timeseries/util/TimeSeriesWritableUtils.java]: Minor issue (probably same thing in RRMDSI I assume) - could use a break here.

File: datavec-api/src/main/java/org/datavec/api/timeseries/util/TimeSeriesWritableUtils.java [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+        INDArray maskArray;
+        if (needMaskArray) {
+            maskArray = Nd4j.ones(details.getMinValues(), details.getMaxTSLength());
+        } else {
+            maskArray = null;
+        }
+
+        //Don't use the global RNG as we need repeatability for each subset (i.e., features and labels must be aligned)
+        Random rng = null;
```
Comment (AlexDBlack) [inline datavec-api/src/main/java/org/datavec/api/timeseries/util/TimeSeriesWritableUtils.java]: Field can be removed - not used here.

File: datavec-api/src/main/java/org/datavec/api/timeseries/util/TimeSeriesWritableUtils.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+
+        for (int i = 0; i < details.getMinValues(); i++) {
+            List<List<Writable>> sequence = list.get(i);
+
+            //Offset for alignment:
+            int startOffset;
+            startOffset = 0;
+            int t = 0;
+            int k;
```
Comment (AlexDBlack) [inline datavec-api/src/main/java/org/datavec/api/timeseries/util/TimeSeriesWritableUtils.java]: Could clean up these - startOffset = 0 always... just need t or k

File: datavec-spark-inference-parent/datavec-spark-inference-model/src/main/java/org/datavec/spark/transform/CSVSparkTransform.java [hunk 4]
```diff
@@ -138,7 +138,7 @@ public SequenceBatchCSVRecord transformSequence(SequenceBatchCSVRecord batchCSVR
      */
     public Base64NDArrayBody transformSequenceArray(SequenceBatchCSVRecord batchCSVRecordSequence) {
         List<List<List<String>>> strings = batchCSVRecordSequence.getRecordsAsString();
-        INDArray arr = Nd4j.create(strings.size(),strings.get(0).size(),strings.get(0).get(0).size());
+        INDArray arr = Nd4j.create(strings.size(),strings.get(0).get(0).size(),strings.get(0).size());
```
Comment (AlexDBlack) [inline datavec-spark-inference-parent/datavec-spark-inference-model/src/main/java/org/datavec/spark/transform/CSVSparkTransform.java]: This looks correct. should be [outer.size, innermost.size, middle.size]

**Pull request merged**